pkgpanda/util.py

```diff
@@ -218,8 +219,13 @@ def extract_tarball(path, target):
         assert os.path.exists(path), "Path doesn't exist but should: {}".format(path)
         make_directory(target)
 
-        with tarfile.open(name=str(path), mode='r') as tar:
-            tar.extractall(path=str(target), numeric_owner=True)
```

Comment:
Do we lose anything by no longer specifying numeric owner anywhere?

Reply:
We might, I guess only CI can answer that question at the moment.

Reply:
And if we find some edge case later we find missing tests - we must be willing to take those risks else we will be paralyzed by the past.

Reply:
The numeric owner option is applied when creating the archive in our case via the `tarfile.TarInfo`.
It is in the archive and the decompression is the same as before so I guess we do not lose anything.

Reply:
Is it? Before we touched anything there was the UID and GID set but there was also an option which forced numeric owner:

```
--numeric-owner
always use numbers for user/group names
```

Reply:
I've checked it by looking into a tar'd DC/OS package (using `dcos-shell` tar) before and after the change.
Before the change there is no non-numeric owner and group information stored:
```
$ tar -tvJf bootstrap--8704a6496c6dfe56980c4926f40e94b31da7f067.tar.xz
-rwxr-xr-x 0/0            2568 2018-03-22 19:45 ./bin/bootstrap
```
Afterwards there is also only numeric information about the owner:
```
$ tar -tvJf bootstrap--8704a6496c6dfe56980c4926f40e94b31da7f067.tar.xz
-rwxr-xr-x 0/0            2568 2019-01-28 17:27 ./bin/bootstrap
```
On extraction if the superuser extracts a tar archive the information in the archive will be used.
Look for `--same-owner` in the manual:
https://www.gnu.org/software/tar/manual/html_node/Attributes.html#SEC138
For non-root users this applies (from the same manual):
```
When restoring, [tar] tries to look the name (if one was written) up in `/etc/passwd'. If it fails, then it uses the user ID stored in the archive instead.
```

Reply:
`bsdtar` has the same semantics if running as `root`:
https://www.freebsd.org/cgi/man.cgi?query=bsdtar&sektion=1&manpath=freebsd-release-ports